tests/integration/targets/server/tasks/test_primary_ips.yml

```diff
@@ -51,32 +51,140 @@
-    name: "{{ hcloud_server_name }}v4"
+    name: "{{ hcloud_primary_ip_name }}v42"
     state: absent
-- name: cleanup test create primary ips
+- name: verify cleanup second primary ipv4
+  assert:
+    that:
+      - result is changed
+- name: cleanup primary ipv6
   hetzner.hcloud.primary_ip:
-    name: "{{ hcloud_server_name }}v42"
+    name: "{{ hcloud_primary_ip_name }}v6"
     state: absent
-  until: result is not failed
+- name: verify cleanup primary ipv6
+  assert:
+    that:
+      - result is changed
+
+- name: test create tertiary ipv4 attached to server with auto_delete
+  hetzner.hcloud.primary_ip:
+    name: "{{ hcloud_primary_ip_name }}v43"
+    server: "{{ hcloud_server_name }}"
+    type: ipv4
+    state: present
+    auto_delete: true
+  register: thirdPrimaryIPv4
+- name: verify test create tertiary ipv4 attached to server with auto_delete
+  assert:
+    that:
+      - thirdPrimaryIPv4 is changed
+- name: test tertiary ipv4 assignee_id with auto_delete
+  hetzner.hcloud.primary_ip:
+    name: "{{ hcloud_primary_ip_name }}v43"
+    state: present
+  register: thirdPrimaryIPv4
+  until: thirdPrimaryIPv4.hcloud_primary_ip.assignee_id is not none
   retries: 5
   delay: 2
-- name: cleanup test create primary ips
+- name: verify test tertiary ipv4 assignee_id with auto_delete
+  assert:
+    that:
+      - thirdPrimaryIPv4.hcloud_primary_ip.assignee_id == server.hcloud_server.id
+      - thirdPrimaryIPv4.hcloud_primary_ip.auto_delete is sameas true
+
+- name: test create secondary ipv6 attached to server
   hetzner.hcloud.primary_ip:
-    name: "{{ hcloud_server_name }}v6"
-    state: absent
-  until: result is not failed
+    name: "{{ hcloud_primary_ip_name }}v62"
+    server: "{{ hcloud_server_name }}"
+    type: ipv6
+    state: present
+    auto_delete: false
+  register: secondPrimaryIPv6
+- name: verify test create secondary ipv6 attached to server
+  assert:
+    that:
+      - secondPrimaryIPv6 is changed
+- name: test secondary ipv6 assignee_id
+  hetzner.hcloud.primary_ip:
+    name: "{{ hcloud_primary_ip_name }}v62"
+    state: present
+  register: secondPrimaryIPv6
+  until: secondPrimaryIPv6.hcloud_primary_ip.assignee_id is not none
   retries: 5
   delay: 2
+- name: verify test secondary ipv6 assignee_id
+  assert:
+    that:
+      - secondPrimaryIPv6.hcloud_primary_ip.assignee_id == server.hcloud_server.id
+      - secondPrimaryIPv6.hcloud_primary_ip.auto_delete is sameas false
+
+- name: cleanup secondary ipv6 attached to server
+  hetzner.hcloud.primary_ip:
+    name: "{{ hcloud_primary_ip_name }}v62"
+    state: absent
+  register: result
+- name: verify cleanup secondary ipv6 attached to server
+  assert:
+    that:
+      - result is changed
+
+- name: cleanup server with primary ips
+  hetzner.hcloud.server:
+    name: "{{ hcloud_server_name }}"
+    state: absent
+  register: server
+  ignore_errors: true
+- name: cleanup server with primary ips - retry due to many timeouts found during testing
```

Comment:
There seems to be a bug that needs investigating.  I am facing many timeouts when I delete a server with a Primary IP set to `auto_delete`.  This is seen both in the pipelines and in my local environment.

In order for the tests to pass I have ignored the error, added a retry, and asserting the server does not exist.

Reply:
See jobs:
- [20240220.7](https://dev.azure.com/ansible/hetzner.hcloud/_build/results?buildId=104619)
- [20240220.6](https://dev.azure.com/ansible/hetzner.hcloud/_build/results?buildId=104613)